### PR TITLE
Feature/cwall 200

### PR DESCRIPTION
**Pull request by sksadjad**
added credential-branding module

### CODE REVIEW

File: packages/credential-branding/src/mapping/CredentialMapper.ts [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+function findCorrelationIdName(correlationId: string, contacts: Party[]): string {
+  const contact = contacts.find((contact: Party) =>
+    contact.identities.some((identity: Identity): boolean => identity.identifier.correlationId === correlationId),
+  )
+
+  if (contact) {
+    return contact.contact.displayName
+  }
+
+  // TODO: bring this back. This checks if the activeUser has the matching correlationId.
```
Comment (maikel-maas) [inline packages/credential-branding/src/mapping/CredentialMapper.ts:21]: Place a link to a ticket in the TODO or remove it

File: packages/credential-branding/src/mapping/CredentialMapper.ts [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+      rows.push({
+        id: uuidv4(),
+        label: key,
+        value: value,
+      })
+      continue
+    }
+
+    if (typeof value !== 'string') {
+      // FIXME disabled this to not show keys of objects
```
Comment (maikel-maas) [inline packages/credential-branding/src/mapping/CredentialMapper.ts:54]: Place a link to a ticket in the TODO or remove it. Remove the commented-out code as well

File: packages/credential-branding/src/mapping/CredentialMapper.ts [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+  const url = typeof verifiableCredential.issuer !== 'string' ? verifiableCredential.issuer.url : undefined
+  const name: string =
+    typeof verifiableCredential.issuer === 'string'
+      ? verifiableCredential.issuer
+      : verifiableCredential.issuer?.name ?? verifiableCredential.issuer?.id
+
+  let issuerAlias: string | undefined = undefined
+  if (typeof verifiableCredential?.issuer === 'object' && verifiableCredential.issuer?.name) {
+    // if the name is part of the VC itself, always use that
+    // todo: Probably still wise to allow a user to override it
```
Comment (maikel-maas) [inline packages/credential-branding/src/mapping/CredentialMapper.ts:151]: Place a link to a ticket in the TODO or remove it

File: packages/credential-branding/src/mapping/CredentialMapper.ts [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+  const localeBranding: IBasicCredentialLocaleBranding | IBasicIssuerLocaleBranding | undefined = args.localeBranding?.find(
+    (branding: IBasicCredentialLocaleBranding | IBasicIssuerLocaleBranding) =>
+      branding.locale?.startsWith(appLocale) || branding.locale === undefined,
+  )
+
+  const logo: string | undefined = localeBranding?.logo?.dataUri || localeBranding?.logo?.uri
+  let imagePreloader: IImagePreloader
+
+  if (typeof navigator !== 'undefined' && navigator.product === 'ReactNative') {
+    //TODO: bring back the react-native related code
```
Comment (maikel-maas) [inline packages/credential-branding/src/mapping/CredentialMapper.ts:193]: Place a link to a ticket in the TODO or remove it.

File: packages/credential-branding/src/utils/MappingUtils.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+/**
+ * @return
+ *  true means the credential is revoked.
+ *  false means the credential is not revoked.
+ */
+export const isRevoked = (): boolean => {
+  return false
+  // TODO implement
```
Comment (maikel-maas) [inline packages/credential-branding/src/utils/MappingUtils.ts:23]: Place a link to a ticket in the TODO or remove it. Remove the commented-out code as well

**Review by maikel-maas — CHANGES_REQUESTED** [review]
Left some remarks

File: packages/credential-branding/src/services/ReactNativeImagePreloader.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import {IPreloadImage, IImagePreloader} from './IPreloadImage'
+/*const FastImage = require('react-native-fast-image')*/
+
+export class ReactNativeImagePreloader implements IImagePreloader {
+  async preload(sources: Array<IPreloadImage>): Promise<void> {
+    /*FastImage.preload(sources)*/
```
[the remark this answers is not in the document]
Comment (nklomp) [reply packages/credential-branding/src/services/ReactNativeImagePreloader.ts:6]: No don't if this is literally the correct code but needs to be moved

File: packages/credential-branding/src/mapping/CredentialMapper.ts (hunk 1 above)
Comment (nklomp) [reply packages/credential-branding/src/mapping/CredentialMapper.ts:21]: No don't if this is literally the correct code but needs to be moved. It needs to live somewhere so the person picking it up knows what the code was and what to do with it. This is entirely different from some old code. We simply have a TODO here

**Review by nklomp — COMMENTED** [review]
@maikel-maas  You are correct that it needs to be linked to a ticket. But this is not a case of removing the code. The ticket would litteraly be about making it work again in RN. Then removing all code occurances, only makes it way harder. This is not some comments of old/dead code that should go

File: packages/credential-branding/LICENSE
Comment (nklomp) [inline packages/credential-branding/LICENSE:1]: What is this license doing here? Should be apache2

File: packages/credential-branding/package.json [hunk 7]
```diff
@@ -0,0 +1,51 @@
+{
+  "name": "@sphereon/ui-components.credential-branding",
+  "private": false,
+  "version": "0.2.0",
+  "description": "SSI UI components credential Branding",
+  "repository": "git@github.com:Sphereon-Opensource/UI-Components.git",
+  "author": "Sphereon <dev@sphereon.com>",
+  "license": "Apache-2.0",
```
Comment (nklomp) [inline packages/credential-branding/package.json:8]: Okay, next time be more careful with licensense. This is mentioning Apache2 as it should, but you included an actual GPL license

File: packages/credential-branding/README.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+<h1 align="center">
+  <br>
+  <a href="https://www.sphereon.com"><img src="https://sphereon.com/content/themes/sphereon/assets/img/logo.svg" alt="Sphereon" width="400"></a>
+  <br>UI Components Credential Branding (Typescript) 
+  <br>
+</h1>
+
+---
+
+**Warning: This package still is in every early development. Breaking changes without notice will happen at this point!**
```
Comment (nklomp) [inline packages/credential-branding/README.md:11]: Remove this line altogether please

File: packages/credential-branding/src/mapping/CredentialMapper.ts (hunk 3 above)
Comment (nklomp) [reply packages/credential-branding/src/mapping/CredentialMapper.ts:151]: Not every todo needs a ticket. It is nice if we create one. But this is a perfect example of a TODO that doesn't necessarily need a ticket. Every FIXME needs one

**Review by nklomp — COMMENTED** [review]
For next time please ring the bell earlier. We have way too many things todo to be working on this a few days and then not have the full solution to begin with. The better approach would have been to discuss creating a copy make it work for web only and create a ticket to refactor it making it available for both. Obviously a shared package should not all of a sudden start creating RN instances etc

File: packages/credential-branding/LICENSE
Comment (sksadjad) [reply packages/credential-branding/LICENSE:1]: yeah replaced it

File: packages/credential-branding/src/mapping/CredentialMapper.ts (hunk 4 above)
Comment (sksadjad) [reply packages/credential-branding/src/mapping/CredentialMapper.ts]: added

File: packages/core/LICENSE [hunk 9]
```diff
@@ -1,636 +1,201 @@
-# GNU GENERAL PUBLIC LICENSE
```
Comment (nklomp) [inline packages/core/LICENSE:1]: Apache2 license

File: packages/core/LICENSE [hunk 10]
```diff
@@ -1,636 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [2023] [Sphereon BV]
```
Comment (nklomp) [inline packages/core/LICENSE:189]:
```suggestion
   Copyright [2024] [Sphereon International B.V.]
```

**Pull request merged**